packages/ethers-5/exports/react.tsx

```diff
@@ -0,0 +1,64 @@
+'use client'
+
+import type { Web3ModalOptions } from '../src/client.js'
+import { Web3Modal } from '../src/client.js'
+import { ConstantsUtil } from '@web3modal/utils'
+import { ProviderController } from '../src/controllers/ProviderController.js'
+import { getWeb3Modal } from '@web3modal/scaffold-react'
+import type { Web3ModalScaffold } from '@web3modal/scaffold'
+import { useSnapshot } from 'valtio'
+
+// -- Types -------------------------------------------------------------------
+export type { Web3ModalOptions } from '../src/client.js'
+
+// -- Setup -------------------------------------------------------------------
+let modal: Web3ModalScaffold | undefined = undefined
+
+export function createWeb3Modal(options: Web3ModalOptions) {
+  if (!modal) {
```

Comment:
Why is modal init here different from what we do in wagmi, in particular using Web3ModalScaffold type explicitly vs Web3Modal?

Reply:
It did this, because of new `getWeb3Modal(modal)` function. We need this function, so that we have `modal` available in shared scaffold packages. `modal` has shared `Web3ModalScaffold` type, so that is why I'm using `Web3ModalScaffold` type in specific `ethers` and `wagmi` packages. 

Do you think there is a better solution for this?

Reply:
We should never expose Web3ModalScaffold (client or type) to end user, hence why each of the exports should only define modal as `Web3Modal`. For wagmi and ethers instance of Web3Modal is different, both extend scaffold modal, but have slightly different options and implementations, so here an ethers specific type should be used. If this means duplicating createWeb3Modal package, its fine, because modal instances that are created will be mostly similar for ethers and wagmi, yet different.